h2o-extensions/target-encoder/src/main/java/ai/h2o/targetencoding/TargetEncoder.java

```diff
@@ -235,11 +235,15 @@ Frame imputeNAsForColumn(Frame data, String teColumnName, String strToImpute) {
     
     private void updateDomainGlobally(Frame fr, String teColumnName, String[] domain) {
       Lockable lock = fr.write_lock();
-      Vec updatedVec = fr.vec(teColumnName);
-      updatedVec.setDomain(domain);
-      DKV.put(updatedVec);
-      fr.update();
-      lock.unlock();
+      try {
+        Vec updatedVec = fr.vec(teColumnName);
+        updatedVec.setDomain(domain);
```

Comment:
don't drop the DKV.put - that is necessary for the changes to propagate to other nodes

Reply:
It's just the update propagation problem to me.

Reply:
@Pscheidl yeah, maybe. Did you have a chance to try this fix in a separate PR, to see results from mr Jenkins?

Reply:
Creating it right now, I'm investigating the lock mechanism for obvious issues. I need approx. 30 minutes.

Reply:
Here is the continuation: https://github.com/h2oai/h2o-3/pull/4501